hologram_dashboard/extension/src/app/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit } from '@angular/core';
+import { RequestHandler } from '../request-handler.component'
+
+export interface DateAvailability {
+  date: string;
+  status: boolean;
+}
+
+export interface HologramDataAvailability {
+  name: string;
+  dates: Array<DateAvailability>
+}
+
+@Component({
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.css'],
+  providers: [ RequestHandler ]
+})
+export class DashboardComponent implements OnInit {
+  dashboard: Array<HologramDataAvailability>;
+  chipper: boolean
+  chipper_gdpr: boolean
+  display: boolean
+
+  constructor(private requestHandler: RequestHandler) { 
+    this.chipper = false;
+    this.chipper_gdpr = false;
+    this.display = false;
+    this.dashboard = []
+  }
+
+  ngOnInit(): void {
+  }
+
+  renderTemplate(hologram_data_availability: JSON, system: string) {
+    for (let key in hologram_data_availability) {
+      let data_set_availability = {} as HologramDataAvailability;
+      data_set_availability.name = key;
+      data_set_availability.dates = [];
+      hologram_data_availability[key].forEach(data => {
+        let date_info = {} as DateAvailability;
+        date_info.date = data[0];
+        date_info.status = data[1];
+        data_set_availability.dates.push(date_info);
+      });
+      this.dashboard.push(data_set_availability);
+    }
+  }
+
+  onClick(system: string) {
+    if (system === "Chipper") {
+      if (this.chipper) {
+        this.chipper = false;
+      } else {
+        this.chipper = true;
+        this.chipper_gdpr = false;
+      }
+    } else if (system === "Chipper_GDPR") {
+      if (this.chipper_gdpr) {
+        this.chipper_gdpr = false;
+      } else {
+        this.chipper_gdpr = true;
+        this.chipper = false;
+      }
```

Comment:
I am a bit confused about this part. If "chipper" button is clicked, then the server would return data for chipper dashboard, right? Why do we need these two boolean variables?

Reply:
it's so that if the user clicks chipper when the dashboard of chipper is showing the program knows to hide the dashboard, similarly, if the user clicks chipper_gdpr when the dashboard of chipper is showing the program knows to delete the dashboard for chipper and show the dashboard for chipper_gdpr instead. With only the display boolean we can't know whether the displayed dashboard is for chipper or chipper_gdpr so we don't know whether we want to hide the dashboard altogether or remove the current dashboard and add the new one.